### PR TITLE
Blueprints resource

### PR DESCRIPTION
**Pull request by hedwigz**

### CODE REVIEW

File: port/cli/blueprint.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return nil, err
+	}
+	if !pb.OK {
+		return nil, fmt.Errorf("failed to create blueprint, got: %s", resp.Body())
+	}
+	return &pb.Blueprint, nil
+}
+
+func (c *PortClient) DeleteBlueprint(ctx context.Context, id string) error {
```
Comment (dvirsegev) [inline port/cli/blueprint.go:49]: You need to do also a UpdateBlueprint?

File: port/cli/permission.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			"permissions": scopes,
+		}).
+		SetResult(&responseBody).
+		Post(url)
+	if err != nil {
+		return err
+	}
+	if statusOK, ok := responseBody["ok"].(bool); ok && !statusOK {
+		return fmt.Errorf("failed to create permissions: %s", resp.Body())
+	} else if !ok {
```
Comment (dvirsegev) [inline port/cli/permission.go:26]: I think is redunand, you can't have status 200 with ok false IMO

File: port/resource_port_blueprint.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+			"data_source": {
+				Type:        schema.TypeString,
+				Description: "The data source for entities of this blueprint",
+				Default:     "Port",
+				Optional:    true,
+			},
+			"icon": {
+				Type:         schema.TypeString,
+				Description:  "The icon of the blueprint",
+				ValidateFunc: validation.StringInSlice([]string{"Actions", "Airflow", "Ansible", "Argo", "AuditLog", "Aws", "Azure", "Blueprint", "Bucket", "Cloud", "Cluster", "CPU", "Customer", "Datadog", "Day2Operation", "DefaultEntity", "DefaultProperty", "DeployedAt", "Deployment", "DevopsTool", "Docs", "Environment", "Git", "Github", "GitVersion", "GoogleCloud", "GPU", "Grafana", "Infinity", "Jenkins", "Lambda", "Link", "Lock", "Microservice", "Moon", "Node", "Okta", "Package", "Permission", "Relic", "Server", "Service", "Team", "Terraform", "User"}, false),
```
Comment (dvirsegev) [inline port/resource_port_blueprint.go:40]: can you set this as const?

File: port/resource_port_blueprint.go [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+	d.Set("relations", &rels)
+}
+
+func writeBlueprintFieldsToResource(d *schema.ResourceData, b *cli.Blueprint) {
+	d.SetId(b.Identifier)
+	d.Set("title", b.Title)
+	d.Set("icon", b.Icon)
+	d.Set("data_source", b.DataSource)
+	d.Set("created_at", b.CreatedAt.String())
+	d.Set("created_by", b.CreatedBy)
```
Comment (dvirsegev) [inline port/resource_port_blueprint.go:176]: CreatedBy should be also in string?

**Review by dvirsegev — APPROVED** [review]
Great job :)

File: port/cli/blueprint.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	url := "v0.1/blueprints/{identifier}"
```
Comment (dvirsegev) [inline port/cli/blueprint.go:50]: maybe we should create an const file for all the urls

File: port/cli/blueprint.go (hunk 1 above)
Comment (hedwigz) [reply port/cli/blueprint.go:49]: good catch, I should also add a test for that

File: port/resource_port_blueprint.go (hunk 4 above)
Comment (hedwigz) [reply port/resource_port_blueprint.go:176]: no, only `time.Time` should be printed with `.String()` (`CreatedBy` is of type string)

File: port/cli/blueprint.go (hunk 5 above)
Comment (hedwigz) [reply port/cli/blueprint.go:70]: Agreed, in the next PR :)

**Pull request merged**